server/models/schemas/player.ts

```diff
@@ -77,12 +77,15 @@ const schema = new Schema({
             score: { type: Types.Number, required: true, default: 0  }
         }
     ],
-    diplomacy: [
-        {
-            playerId: { type: Types.ObjectId, required: true },
-            status: { type: Types.String, required: true }
-        }
-    ]
+    diplomacy: {
+        otherPlayers: [ 
+          {
+              playerId: { type: Types.ObjectId, required: true },
+              status: { type: Types.String, required: true }
+          }
+        ],
+        alliancesMadeThisCycle : { type: Types.Number, required: true, default: 0 }
```

Comment:
This feels yucky, not a fan. I'll investigate other ways of achieving this without needing to store and keep this value up to date.